src/usps_processor/parse.clj

```diff
@@ -33,24 +34,117 @@
   "Takes the time in the USPS string format, and a timezone id like
    'America/New_York', and returns a DateTime in UTC for that time."
   [time-string timezone-id]
-  (let [timezone (timezone-id->timezone timezone-id)
-        formatter (format/formatter usps-time-format-pattern timezone)
-        time-in-zone (format/parse formatter time-string)]
-    (time/to-time-zone time-in-zone time/utc)))
+  (try
+    (let [timezone (timezone-id->timezone timezone-id)
+          formatter (format/formatter usps-time-format-pattern timezone)
+          time-in-zone (format/parse formatter time-string)]
+      (time/to-time-zone time-in-zone time/utc))
+    (catch Exception ex
+      (log/warn "Exception parsing scan time" ex))))
+
+(defn datetime->date
+  "Converts a datetime to date, or returns nil if datetime is nil"
+  [datetime]
+  (when datetime
+    (coerce/to-date datetime)))
```

Comment:
Yeah, it's just a thin wrapper so as not to try and coerce nil to a date. It allows the nil to be carried forward, which is then checked against the schema validation which says the `:scan-time` needs to be a j.u.Date.